logbook-logstash-logback-encoder/src/main/java/org/zalando/logbook/lle/AutodetectPrettyPrintingMarker.java

```diff
@@ -0,0 +1,45 @@
+package org.zalando.logbook.lle;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.PrettyPrinter;
+
+import net.logstash.logback.marker.RawJsonAppendingMarker;
+
+/**
+ * 
+ * Auto-detecting pretty-printing Marker. Pretty-printing cannot be performed in advance.
+ *
+ */
+
+public class AutodetectPrettyPrintingMarker extends RawJsonAppendingMarker {
+    
+    private static final long serialVersionUID = 1L;
+
+    public AutodetectPrettyPrintingMarker(String fieldName, String rawJson) {
+        super(fieldName, rawJson);
+    }
+
+    @Override
+    protected void writeFieldValue(JsonGenerator generator) throws IOException {
+        PrettyPrinter prettyPrinter = generator.getPrettyPrinter();
+        if(prettyPrinter == null) {
+            super.writeFieldValue(generator);
+        } else {
+            // append to existing tree event by event
+            final JsonParser parser = generator.getCodec().getFactory().createParser((String)super.getFieldValue());
```

Comment:
The logbook-json module contains a pretty printing body filter. If that one is used in conjunction you can ignore it here and just use `generator.writeRaw(..)`.

Reply:
That will not work out of the box, because the indent level will one level off. But I'll look into it.

Reply:
Actually I think the marker approach is more tidy as it uses the log framework's prettyprinting. There might be a case for improved performance with pretty-printing here, but thats not for production use anyways, so it is not that important (at least not to us). 

I'm not claiming that very many are doing this, but it also keeps the door open for stuff like this:

https://github.com/skjolber/logback-logstash-syntax-highlighting-decorators

Reply:
Agreed.